src/lib/services/api-token-service.ts

```diff
@@ -63,6 +74,12 @@ export class ApiTokenService {
             () => this.fetchActiveTokens(),
             minutesToMilliseconds(1),
         ).unref();
+        if (this.flagResolver.isEnabled('tokensLastSeen')) {
+            this.seenTimer = setInterval(
+                () => this.updateLastSeen(),
```

Comment:
Should we await here? Something like:
```suggestion
                () => await this.updateLastSeen(),
```
The concern is: if `updateLastSeen` executes asynchronously, some updateLastSeen might overlap over time (i.e. run 2 in parallel). It'd take quite a lot of time to happen with 1 minute between operations, but I've seen this before. 

I usually wait until the previous operation finishes before triggering the timer again. I'm not sure how this is done in node (or any best practice). Just wanted to touch base on this but it's not a blocker

Reply:
I don't think it works like this, as it still doesn't ensure sequentiality, but I may be wrong.
What do you think about an approach like https://github.com/Unleash/unleash/pull/2514/commits/47a1f0a8b8da574c23253076b1941e6231b516b6 inspired by: https://github.com/Unleash/unleash/blob/a992aca228c3bc0c17abcac22d027796db8059c0/src/lib/proxy/proxy-repository.ts#L91-L97

Reply:
Yes, I read a bit about the topic and your change is in line with my intuition: setTimeout is less risky (operationally speaking) cause there's no risk of executing multiple instances of the code concurrently. It does lag if the function takes longer, but that's exactly what I'd like to see in this particular use case: to execute less frequently in situations of high CPU load or other situations that may cause this function to execute slower.

Resources:
- https://blog.skay.dev/when-to-use-settimeout-vs-setinterval#conclusion
- https://stackoverflow.com/questions/729921/settimeout-or-setinterval